src/main/java/org/jvnet/hudson/plugins/platformlabeler/NodeLabelCache.java

```diff
@@ -62,8 +62,10 @@
   @Override
   public final void onOnline(final Computer computer, final TaskListener ignored)
       throws IOException, InterruptedException {
-    cacheLabels(computer);
-    refreshModel(computer);
+    if (computer.getNumExecutors() > 0) {
```

Comment:
I personally prefer having all nodes labeled, whether or not they have an executor at that moment.  However, if you feel strongly that a label should not be applied to a master with 0 executors, then I think we should make the code explicitly declare that by checking for the name of the node.

```suggestion
    if (computer.getNumExecutors() > 0 && computer.getName().equals("master")) {
```